SimG4CMS/Forward/src/BHMNumberingScheme.cc

```diff
@@ -6,43 +6,20 @@
 
 BHMNumberingScheme::BHMNumberingScheme() { LogDebug("BHMSim") << " Creating BHMNumberingScheme"; }
 
-int BHMNumberingScheme::detectorLevel(const G4Step* aStep) const {
-  //Find number of levels
-  const G4VTouchable* touch = aStep->GetPreStepPoint()->GetTouchable();
-  return (touch) ? ((touch->GetHistoryDepth()) + 1) : 0;
-}
-
-void BHMNumberingScheme::detectorLevel(const G4Step* aStep, int& level, int* copyno, G4String* name) const {
-  //Get name and copy numbers
-  if (level > 0) {
-    const G4VTouchable* touch = aStep->GetPreStepPoint()->GetTouchable();
-    for (int ii = 0; ii < level; ++ii) {
-      int i = level - ii - 1;
-      name[ii] = ForwardName::getName(touch->GetVolume(i)->GetName());
-      copyno[ii] = touch->GetReplicaNumber(i);
-    }
-  }
-}
-
 unsigned int BHMNumberingScheme::getUnitID(const G4Step* aStep) const {
   unsigned intindex = 0;
-  int level = detectorLevel(aStep);
 
-  LogDebug("BHMSim") << "BHMNumberingScheme number of levels= " << level;
-  if (level > 0) {
-    int* copyno = new int[level];
-    G4String* name = new G4String[level];
-    detectorLevel(aStep, level, copyno, name);
+  //Find number of levels
+  const G4VTouchable* touch = aStep->GetPreStepPoint()->GetTouchable();
+  int level = (touch) ? ((touch->GetHistoryDepth()) + 1) : 0;
```

Comment:
I do not think that touch can be undefined.

Reply:
The check was there in the [original code](https://github.com/cms-sw/cmssw/pull/41942/files/3c866ea649c4da1c3dc5ba483057eb7307f5c1cf#diff-5b5049e9d92e98458c2acbd42d3ea28d19afb6062dd73ea2629c3e68530c37f8L12), I can drop it if you think it's safe.